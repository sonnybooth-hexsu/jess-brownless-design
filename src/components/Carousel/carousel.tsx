import React, { ReactNode } from "react"
import { ArrowRight, ArrowLeft } from "react-feather"
import Slider from "react-slick"
import "./slick-overrides.module.css"

type ArrowProps = {
  className?: string
  style?: object
  onClick?: () => void
}

type CarouselProps = {
  settingsOverrides?: object
  children: ReactNode
}

const NextArrow = ({ className, style, onClick }: ArrowProps) => {
  return (
    <div className={className} style={{ ...style }} onClick={onClick}>
      <ArrowRight />
    </div>
  )
}

const PrevArrow = ({ className, style, onClick }: ArrowProps) => {
  return (
    <div className={className} style={{ ...style }} onClick={onClick}>
      <ArrowLeft />
    </div>
  )
}

export const Carousel = ({ settingsOverrides, children }: CarouselProps) => {
  const settings = {
    dots: true,
    infinite: false,
    speed: 500,
    slidesToScroll: 1,
    slidesToShow: 2.5,
    swipeToSlide: true,
    nextArrow: <NextArrow />,
    prevArrow: <PrevArrow />,
    responsive: [
      {
        breakpoint: 1024, //Tailwind lg default
        settings: {
          slidesToShow: 2,
        },
      },
      {
        breakpoint: 768, //Tailwind md default
        settings: {
          slidesToShow: 1,
        },
      },
    ],
    ...settingsOverrides,
  }

  return <Slider {...settings}>{children}</Slider>
}
