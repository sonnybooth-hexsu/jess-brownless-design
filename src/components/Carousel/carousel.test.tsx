/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { queryAllByTestId } from "@testing-library/dom"
import { Carousel } from "./carousel"

afterEach(() => {
  cleanup()
})

describe("<ProductCard />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Carousel>
        <div>Carousel Item 1</div>
        <div>Carousel Item 2</div>
        <div>Carousel Item 3</div>
        <div>Carousel Item 4</div>
      </Carousel>
    )

    expect(container).toMatchSnapshot()
  })

  it("number of children/slides passed through should match", () => {
    const { container } = render(
      <Carousel>
        <div data-testid="carouselSlider">Carousel Item 1</div>
        <div data-testid="carouselSlider">Carousel Item 2</div>
        <div data-testid="carouselSlider">Carousel Item 3</div>
        <div data-testid="carouselSlider">Carousel Item 4</div>
      </Carousel>
    )

    expect(queryAllByTestId(container, "carouselSlider")).toHaveLength(4)
  })
})
