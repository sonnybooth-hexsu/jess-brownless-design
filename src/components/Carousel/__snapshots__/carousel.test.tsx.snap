// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should match snapshot 1`] = `
<div>
  <div
    class="slick-slider slick-initialized"
    dir="ltr"
  >
    <div
      class="slick-arrow slick-prev slick-disabled"
      style="display: block;"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="19"
          x2="5"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 19 5 12 12 5"
        />
      </svg>
    </div>
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              Carousel Item 1
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="1"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              Carousel Item 2
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="2"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              Carousel Item 3
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="3"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              Carousel Item 4
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="slick-arrow slick-next"
      style="display: block;"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
        <polyline
          points="12 5 19 12 12 19"
        />
      </svg>
    </div>
    <ul
      class="slick-dots"
      style="display: block;"
    >
      <li
        class="slick-active"
      >
        <button>
          1
        </button>
      </li>
      <li
        class=""
      >
        <button>
          2
        </button>
      </li>
      <li
        class=""
      >
        <button>
          3
        </button>
      </li>
    </ul>
  </div>
</div>
`;
